eyecite/resolve.py

```diff
@@ -46,9 +57,41 @@ def _filter_by_matching_antecedent(
     return matches[0] if len(matches) == 1 else None
 
 
+def _invalid_pin_cite(full_cite: FullCitation, id_cite: IdCitation):
```

Comment:
Three things:

1. refactor to be `_valid_pin_cite` instead of `_invalid_pin_cite`? These kinds of methods are always more intuitive to me when written "positively." E.g., returning `False` from this method create a confusing (to me) double negative.
2. add `is` prefix, so `_is_invalid_pin_cite`? Again, just improve readability, at least for me.
3. add `-> bool` return signature?

Reply:
I pushed a commit to rename to `_has_invalid_pin_cite` and add the missing return type.

I agree that double negatives are confusing, but I tried `_valid_pin_cite` and found that more confusing, because it also returns true for id cites without pin cites at all, and for full cites without page groups ... all we're really saying here is this id cite doesn't (a) have a pin cite that (b) fits a particular pattern we think is invalid.